### PR TITLE
Performed refactoring of the network service in front end

### PR DESCRIPTION
**Pull request by JayC-SF**
Fixed some typos and performed some refactoring in the code & some bug fixes

### CODE REVIEW

File: frontend/src/hooks/useUserInfo.ts [hunk 1]
```diff
@@ -5,7 +5,7 @@ import { UserInfo } from "../contexts/userinfo";
 /**
  * User info hook that queries the user's information upon the first rendering 
  */
-export function userUserInfo() {
+export function useUserInfo() {
```
Comment (LamdaDev) [inline frontend/src/hooks/useUserInfo.ts:8]: Good catch, was wondering why userUserInfo() was named like that

File: frontend/src/network/http/jsonRequest.ts [hunk 2]
```diff
@@ -32,15 +32,14 @@ export async function JsonRequest(url: string, options: JsonRequestInit = {}) {
...
     const text = await response.text()
     if (text.length > 0) {
         // If text is not empty, parse it as JSON
-        return JSON.parse(text) as Json
+        return JSON.parse(text)
     }
     // no body in the response, let caller handle the data
-    return null
-
+    return undefined
```
Comment (LamdaDev) [inline frontend/src/network/http/jsonRequest.ts]: Any reason why this should return undefined instead of null?

**Review by LamdaDev — APPROVED** [review]
LGTM!

File: frontend/src/network/http/jsonRequest.ts (hunk 2 above)
Comment (JayC-SF) [reply frontend/src/network/http/jsonRequest.ts]: good catch honestly I was trying things out and forgot to revert it back

**Pull request merged**